File: leetcode/array/inplace_zeros/python/inplace_zeros.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nzeros = 0
        write_indx = None
        for read_indx, current_num in enumerate(nums):
            if current_num == 0:
                nzeros += 1
                if write_indx is None:
                    write_indx = read_indx
            else:
                if write_indx is not None:
                    nums[write_indx] = current_num
                    write_indx += 1
        if write_indx is not None:
            for indx in range(write_indx, write_indx + nzeros):
                nums[indx] = 0

File: leetcode/array/inplace_zeros/c/inplace_zeros.c
void moveZeroes(int* nums, int numsSize){
    int read_indx = 0, nzeros = 0, write_indx = -1;
    for (; read_indx < numsSize; read_indx++) {
        int current_num = nums[read_indx];
        if (!current_num) {
            nzeros++;
            if (write_indx == -1) {
                write_indx = read_indx;
            }
        } else {
            if (write_indx != -1) {
                nums[write_indx] = current_num;
                write_indx++;
            }
        }
    }
    if (write_indx != -1) {
        while (nzeros--) {
            nums[write_indx + nzeros] = 0;
        }
    }
}
